src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -1,32 +1,122 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Collections.Generic;
 using System.CommandLine.Parsing;
+using System.IO;
 using System.Linq;
+using System.Text.Json;
 using Microsoft.DotNet.Cli;
 using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
 
 namespace Microsoft.DotNet.Workloads.Workload.Install
 {
     internal class WorkloadInstallCommand : CommandBase
     {
-        private readonly string _framework;
-        private IReadOnlyCollection<string> _workloadIds;
+        private readonly string _fromCacheOption;
+        private readonly bool _printDownloadLinkOnly;
+        private readonly IReadOnlyCollection<string> _workloadIds;
+
+        public readonly string MockInstallDirectory = Path.Combine(CliFolderPathCalculator.DotnetUserProfileFolderPath,
+            "DEV_mockworkloads");
 
         public WorkloadInstallCommand(
             ParseResult parseResult)
             : base(parseResult)
         {
-            _framework = parseResult.ValueForOption<string>(WorkloadInstallCommandParser.FrameworkOption);
-            _workloadIds = parseResult.ValueForArgument<IReadOnlyCollection<string>>(WorkloadInstallCommandParser.WorkloadIdArgument);
+            _workloadIds =
+                parseResult.ValueForArgument<IReadOnlyCollection<string>>(WorkloadInstallCommandParser
+                    .WorkloadIdArgument);
+            _printDownloadLinkOnly =
+                parseResult.ValueForOption<bool>(WorkloadInstallCommandParser.PrintDownloadLinkOnlyOption);
+            _fromCacheOption = parseResult.ValueForOption<string>(WorkloadInstallCommandParser.FromCacheOption);
         }
 
         public override int Execute()
         {
             // TODO stub
-            Reporter.Output.WriteLine($"WIP workload install {string.Join("; ",_workloadIds)}");
+            Reporter.Output.WriteLine($"WIP workload install {string.Join("; ", _workloadIds)}");
+            List<string> allowedMockWorkloads = new List<string> {"mobile-ios", "mobile-android"};
+
+            if (!_printDownloadLinkOnly && string.IsNullOrWhiteSpace(_fromCacheOption))
+            {
+                Reporter.Output.WriteLine(
+                    "mock currently does not support normal install. Need --print-download-link-only or --from-cache");
+            }
+
+            if (_workloadIds.Except(allowedMockWorkloads).Any())
+            {
+                Reporter.Output.WriteLine("Only support \"mobile-ios\", \"mobile-android\", \"mobile\" in the mock");
```

Comment:
nit: `mobile` is listed as supported here but isn't in allowedMockWorkloads

Reply:
removed mobile part. It is complicated, and VSMac doesn't really need to deal with that